src/gc/windows/gcenv.windows.cpp

```diff
@@ -604,8 +1098,184 @@ bool GCToOSInterface::GetWriteWatch(bool resetState, void* address, size_t size,
+
+                        DWORD dwL2CacheBits = dwBuffer[2];
+                        DWORD dwL3CacheBits = dwBuffer[3];
+
+                        maxTrueSize = (size_t)((dwL2CacheBits >> 16) * 1024);    // L2 cache size in ECX bits 31-16
+                                
+                        getcpuid(0x1, buffer);
+                        DWORD dwBaseFamily = (dwBuffer[0] & (0xF << 8)) >> 8;
+                        DWORD dwExtFamily  = (dwBuffer[0] & (0xFF << 20)) >> 20;
+                        DWORD dwFamily = dwBaseFamily >= 0xF ? dwBaseFamily + dwExtFamily : dwBaseFamily;
+
+                        if (dwFamily >= 0x10)
+                        {
+                            BOOL bSkipAMDL3 = FALSE;
+
+                            if (dwFamily == 0x10)   // are we running on a Barcelona (Family 10h) processor?
+                            {
+                                // check model
+                                DWORD dwBaseModel = (dwBuffer[0] & (0xF << 4)) >> 4 ;
+                                DWORD dwExtModel  = (dwBuffer[0] & (0xF << 16)) >> 16;
+                                DWORD dwModel = dwBaseFamily >= 0xF ? (dwExtModel << 4) | dwBaseModel : dwBaseModel;
+
+                                switch (dwModel)
+                                {
+                                    case 0x2:
+                                        // 65nm parts do not benefit from larger Gen0
+                                        bSkipAMDL3 = TRUE;
+                                        break;
+
+                                    case 0x4:
+                                    default:
+                                        bSkipAMDL3 = FALSE;
+                                }
+                            }
+
+                            if (!bSkipAMDL3)
+                            {
+                                // 45nm Greyhound parts (and future parts based on newer northbridge) benefit
+                                // from increased gen0 size, taking L3 into account
+                                getcpuid(0x80000008, buffer);
+                                DWORD dwNumberOfCores = (dwBuffer[2] & (0xFF)) + 1;     // NC is in ECX bits 7-0
+
+                                DWORD dwL3CacheSize = (size_t)((dwL3CacheBits >> 18) * 512 * 1024);  // L3 size in EDX bits 31-18 * 512KB
+                                // L3 is shared between cores
+                                dwL3CacheSize = dwL3CacheSize / dwNumberOfCores;
+                                maxTrueSize += dwL3CacheSize;       // due to exclusive caches, add L3 size (possibly zero) to L2
+                                                                    // L1 is too small to worry about, so ignore it
+                            }
+                        }
+
+
+                        maxSize = maxTrueSize;
+                    }
+                }
+            }
+        }
+    } 
+    __except(EXCEPTION_EXECUTE_HANDLER)
+    {
+        
```

Comment:
Not porting the whole `PAL_TRY` is fine since this is Windows-specific.

It should be fine to just remove the whole `__try/__except`. It was needed to support very old processors that did not have cpuid instruction. We do not support those processors anymore.